apps/examples/media_tash/media_tash_main.c

```diff
@@ -0,0 +1,420 @@
+/****************************************************************************
+ *
+ * Copyright 2016 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <dirent.h>
+#include <string.h>
+#include <apps/shell/tash.h>
+#include <fcntl.h>
+
+#include <media/media_player.h>
+#include <media/media_recorder.h>
+#include <media/media_utils.h>
+
+#include <dm/dm_connectivity.h>
+#include <dm/dm_error.h>
+#include <sys/socket.h>
+
+#define DISCONNECT_WIFI_AND_CLOSE_SOCKET() do { dm_conn_wifi_disconnect(); disconnect_socket(&fd); } while(0)
+
+int isWiFiConnected = 0;
+
+void WiFi_Link_Up(void)
+{
+	isWiFiConnected = 1;
+	printf("\n[WiFi] Connected!\n");
+}
+
+void WiFi_Link_Down(void)
+{
+	isWiFiConnected = 0;
+	printf("\n[WiFi] Disconnected!\n");
+}
+
+int network_setup(void)
+{
+	isWiFiConnected = 0;
+
+	dm_cb_register_init();
+	dm_conn_wifi_connect(WiFi_Link_Up, WiFi_Link_Down);
+
+	while (true) {
+		if (isWiFiConnected == 1) {
+			break;
+		}
+		printf("[WiFi] Waiting Connect......\n");
+		sleep(1);
+	}
+	dm_conn_dhcp_init();
+
+	if (isWiFiConnected == 1) {
+		return 0;
+	} else {
```

Comment:
please remove else.

Reply:
if isWifiConnected is not set, above while will be run forever.
So currently this if-else is not work properly.
IF you want use this if-else part you can add some try count inside of while loop